docs/guides/migration_guide_packet.md

```diff
@@ -0,0 +1,152 @@
+---
+page_title: "Migration Guide from terraform-provider-packet"
+description: |-
+  Migrating your templates from packet_ resources to metal_ resources with minimal disruption.
+---
+
+# Migrating from packethost/packet to equinix/metal
+
+Packet is now Equinix Metal, and the name of the Terraform provider changed too. This (terraform-provider-metal, provider equinix/metal) is the current provider for Equinix Metal.
+
+If you've been using terraform-provider-packet, and you want to use a newer provider version to manage resources in Equinix Metal, you will need to change the references in you hcl files. You can just change the names of the resources, e.g. from `packet_device` to `metal_device`. That should work, but it will cause the `packet_device` to be destroyed and new `metal_device` to be created instead. Re-creation of the resources might be undesirable, and this guide shows how to migrate to metal_ resources without the re-creation. We can use terraform state and import to achieve transition without destroying users' resources in Equinix Metal.
+
+## Existing template
```

Comment:
Before starting, they should update to the latest packet provider, to get around any other migration issues, like packet_device_network_type, hardware reservation id (if that would be an issue).

Reply:
Maybe they should update to the latest Terraform version first, too (we don't have to describe how this is done, just advise them to do so and follow Hashicorps guides on that).

Or maybe upgrading Terraform versions complicates things and we don't need that?

Will users run into problems if they don't upgrade first? Their is some minimal version that is needed before using out of tree providers without manual install, right? Was that 0.11 or 0.12?

Reply:
I think the tree providers was 0.13. https://www.terraform.io/upgrade-guides/0-13.html#explicit-provider-source-locations